### PR TITLE
Missing system packages

### PR DESCRIPTION
**Pull request by tobiashochguertel**
I installed gitlab with the following parameters:

  	## Configure gitlab
  	class {
	 'gitlab':
		gitlab_ruby_version  => '2.1.5',
		gitlab_branch        => '7-9-stable',
		gitlabshell_branch   => 'v2.6.0',
	}

Operation System: Ubuntu 14.04

I hope this helps. It's my first pull-request.

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -75,7 +76,9 @@
     'Debian': {
       # system packages
       $system_packages = ['libicu-dev', 'python2.7','python-docutils',
-                          'libxml2-dev', 'libxslt1-dev','python-dev']
+                          'libxml2-dev', 'libxslt1-dev','python-dev', 'cmake', 'pkg-config', 'libkrb5-dev', 'ruby-execjs']
+                          
+      $redis_packages = ['redis-server']
```
Comment (blkperl) [inline manifests/params.pp]: Redis should be managed outside of this module. There are many Redis modules available on the Puppet Forge.

Comment (tobiashochguertel) [inline manifests/params.pp]: @blkperl I suggest, I had not thought of before. Is there an hint for choose of an dependency?

The following redis module looks like the most used module: https://forge.puppetlabs.com/thomasvandoren/redis

**Comment by tobiashochguertel** [comment]
some changes based on @blkperl informations. I hope this was expected todo.

**Comment by sbadia** [comment]
Hi Tobias, thanks for the patch I'll test this :)

**Comment by sbadia** [comment]
I'll shash your commits

**Comment by tobiashochguertel** [comment]
Hi Sebastien,

do you mean with stash (typo = shash)?

Is that good or bad. What can I do better?

Give me some hint If my contribute is failling. It looks so hard for me to contribute to github.com projects.

**Comment by atomaka** [comment]
@tobiashochguertel he more likely means "squash" instead of "shash."

**Comment by sbadia** [comment]
Hi,
Yes sorry, I mean `squash` :innocent:
It's only in order to keep the GIT history clean. I just merged your commit in a single commit. Using these commands.

A cool full How-To is available here:
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

Ps: here `origin` refer to https://github.com/sbadia/puppet-gitlab.git

So, I fetch your changes, and use the rebase interactive command.

```shell
git remote add tobias github.com:tobiashochguertel/puppet-gitlab.git
git fetch -v tobias
git checkout tobias/missing_system_packages -b missing_system_packages
git rebase -i master
```
The interactive command launch a EDITOR whith this: (a summary of all commit
involved in the rebase).

```shell
pick b1f2635 Missing System Package: 'cmake' when using Ruby-Version 2.1.5, gitlab version: 7.9-stable and gitshell v2.6.0
pick 4ba37de Missing System Package: 'pkg-config' when using Ruby-Version 2.1.5, gitlab version: 7.9-stable and gitshell v2.6.0
pick bc822a4 Missing System Package: 'libkrb5-dev' when using Ruby-Version 2.1.5, gitlab version: 7.9-stable and gitshell v2.6.0
pick a7d016d Missing System Package: 'ruby-execjs' when using Ruby-Version 2.1.5, gitlab version: 7.9-stable and gitshell v2.6.0
pick ee97572 Makes gitlab modules able to manage redis server installation.
pick 1f23e69 redis dependency added. removed optio to self manage redis.
pick 6be30a8 typo fix.
pick 4a7c27c removing option: gitlab_manage_redis
pick d44246a removing redis setup.pp
pick b834a9a removing depency redis.
```
You can use this keywords to edit / rewrite the history.

```
# Commands:
#  p, pick = use commit
#  r, reword = use commit, but edit the commit message
#  e, edit = use commit, but stop for amending
#  s, squash = use commit, but meld into previous commit
#  f, fixup = like "squash", but discard this commit's log message
#  x, exec = run command (the rest of the line) using shell
```

In our case, we just `reword` the first commit (in order to add the others packages), and we `fixup` the commit about redis (because this change was removed after).

The final output/summary look like this :

```shell
r b1f2635 Missing System Package: 'cmake' when using Ruby-Version 2.1.5, gitlab version: 7.9-stable and gitshell v2.6.0
f 4ba37de Missing System Package: 'pkg-config' when using Ruby-Version 2.1.5, gitlab version: 7.9-stable and gitshell v2.6.0
f bc822a4 Missing System Package: 'libkrb5-dev' when using Ruby-Version 2.1.5, gitlab version: 7.9-stable and gitshell v2.6.0
f a7d016d Missing System Package: 'ruby-execjs' when using Ruby-Version 2.1.5, gitlab version: 7.9-stable and gitshell v2.6.0
f ee97572 Makes gitlab modules able to manage redis server installation.
f 1f23e69 redis dependency added. removed optio to self manage redis.
f 6be30a8 typo fix.
f 4a7c27c removing option: gitlab_manage_redis
f d44246a removing redis setup.pp
f b834a9a removing depency redis.
```

With a `reword` of the first commit (b1f2635), with this text:

```
Missing System Package: 'cmake', 'pkg-config', 'libkrb5-dev', 'ruby-execjs'.

When using Ruby-Version 2.1.5, gitlab version: 7.9-stable and gitshell v2.6.0.
puppet-gitlab lacks this system packages.
```

And finaly I just pushed in a branch on my `remote` (https://github.com/sbadia/puppet-gitlab.git)

```shell
git push origin missing_system_packages
```

https://github.com/sbadia/puppet-gitlab/tree/missing_system_packages

**Comment by sbadia** [comment]
Replaced by #216 thanks @tobiashochguertel !

**Comment by tobiashochguertel** [comment]
@sbadia for short, because I wasn't at home the last 4 days. Thank you for your good and complete answer. I'm fascinated about your complete answer, this will help me to get an step closer to an contributor of this and other open source projects.

I'm really happy to get the feedback, that my contribution was successfully :+1: for that! That's my fist time :)

I look closer to your answer in the next days. I want to contribute more because I get answers to questions what really matters in software-developing and how to achieve that :100: . Like an clean history - I know that, but pratically it's hard to figure out how does an clean history looks like.

**Pull request closed without merging**